padl/transforms.py

```diff
@@ -1760,520 +1844,554 @@ def _add_name_to_splits(self, final_splits):
+            if self._check_edge_compatibility(node_b_output_slice, input_slice, parent_position):
+                temp_output_slice = node_b_output_slice if temp_output_slice is None else temp_output_slice
+                temp_output_slice = None if temp_output_slice == parent_position else temp_output_slice
+                self.connect(node_a,
+                             node_b,
+                             output_slice=temp_output_slice,
+                             input_slice=input_slice,
+                             )
+
+        temp_edges = defaultdict(dict)
+        temp_parents = defaultdict(list)
+
+        for parent, children in graph_b.edges.items():
+            if parent in (graph_b.input_node, graph_b.output_node):
+                continue
+            for child in children:
+                if child not in (graph_b.input_node, graph_b.output_node):
+                    temp_edges[parent][child] = graph_b.edges[parent][child]
+        for parent, children in self.edges.items():
+            for child in children:
+                temp_edges[parent][child] = self.edges[parent][child]
+
+        for child, parents in graph_b.parents.items():
+            if child in (graph_b.input_node, graph_b.output_node):
+                continue
+            for parent in parents:
+                if parent in (graph_b.input_node, graph_b.output_node):
+                    continue
+                temp_parents[child].append(parent)
+        for child, parents in self.parents.items():
+            for parent in parents:
+                temp_parents[child].append(parent)
+
+        self.edges = temp_edges
+        self.parents = temp_parents
+
+    @lru_cache
+    def sorted_nodes(self):
+        """Topologically sorted nodes"""
+        return _topological_node_sort(self.input_node, self.edges, self.parents)
+
+    @lru_cache
+    def _set_output_formatter(self):
+        transforms = [node.transform for node in self.parents[self.output_node]]
+        self.pd_keys = self._pd_get_keys(transforms)
+        return self.pd_keys
 
-        Example:
-             t = prep >> batch >> forward_1 >> forward_2
+    def _pd_format_output(self, x):
+        return self._pd_output_formatter(x)
 
-            Let's suppose we get an error on `forward_2`. `t.pd_forward` is
-            `t.pd_forward` = forward_1 >> forward_2`. `forward_2` is the index 1 of `t.pd_forward`,
-            then the element that fails on `t` is
-            2 + t.pd_forward._pd_get_stage_idx() = 2 + 1 = 3
+    def _pd_output_formatter(self, args):
+        keys = self._set_output_formatter()
+        if len(keys) == 1:
+            return args
+        return namedtuple('namedtuple', keys)(*args)
 
-        :param is_entire_transform: *False* if *self* is a part of a larger :class:`Transform`,
-            else *True*
-        """
-        return 0 if self._pd_group and not is_entire_transform else _pd_trace[-1].error_position
+    def __call__(self, args):
```

Comment:
@eramdiaz I think this `__call__` method will need to be updated with a `try` statement to get it to work the debugger. Could you take a look at this and help Jason with the changes?

Reply:
yes, let me know when we can check this @jasonkhadka